### PR TITLE
Improve cluster integrity sync reliability

### LINKED ISSUE #14742: Migrate cluster Integrity sync enhancements to 4.3
## Description

Large environments can see their performance severely affected by timeout errors during file synchronization. 

In those cases, the synchronization task is canceled and started again a little later, but the files continue to be sent even if the worker does not use them because they are associated with a canceled task. Multiple improvements to address this issue were introduced in the following issue:
- https://github.com/wazuh/wazuh/issues/11945

However, these changes will not be applied until Wazuh version v4.4. For said reason, it is intended to migrate everything done in this issue to Wazuh 4.3, so that the problem is solved as soon as possible.

Testing and migrating system tests will be required to ensure the cluster still works properly.

## Checks

The following elements have been updated or reviewed (should also be checked if no modification is required):
- [x] Tests (unit tests, API integration tests).
- [x] Changelog.
- [x] Documentation.
- [x] Integration test mapping (using `api/test/integration/mapping/_test_mapping.py`).

### PR DESCRIPTION
**Pull request by Selutario**
|Related issue|
|---|
| Closes #14742 |

## Description

This PR is a backport of the development that was done in the one below. Refer to it to see all the details:
- https://github.com/wazuh/wazuh/pull/12504

## Logs

These are cluster logs with the backported Integrity sync improvements, copying the conditions (although with only 2 workers) of a real environment that usually has problems due to the large number of multigroups. There are 2214 files whose size is greater than 2 GB altogether:
```
2022/08/26 11:09:44 INFO: [Worker] [Main] Connection from ('192.168.16.5', 37708)
2022/08/26 11:09:45 INFO: [Worker] [Main] Connection from ('192.168.16.6', 45806)
2022/08/26 11:09:48 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:09:48 INFO: [Master] [Local integrity] Finished in 0.135s. Calculated metadata of 2248 files.
2022/08/26 11:09:53 INFO: [Worker worker1] [Integrity check] Starting.
2022/08/26 11:09:53 INFO: [Worker worker1] [Integrity check] Finished in 0.004s. Received metadata of 34 files. Sync required.
2022/08/26 11:09:53 INFO: [Worker worker1] [Integrity sync] Starting.
2022/08/26 11:09:53 INFO: [Worker worker1] [Integrity sync] Files to create in worker: 2214 | Files to update in worker: 0 | Files to delete in worker: 0 | Files to receive: 0
2022/08/26 11:09:54 INFO: [Worker worker2] [Integrity check] Starting.
2022/08/26 11:09:54 INFO: [Worker worker2] [Integrity check] Finished in 0.004s. Received metadata of 34 files. Sync required.
2022/08/26 11:09:54 INFO: [Worker worker2] [Integrity sync] Starting.
2022/08/26 11:09:54 INFO: [Worker worker2] [Integrity sync] Files to create in worker: 2214 | Files to update in worker: 0 | Files to delete in worker: 0 | Files to receive: 0
2022/08/26 11:09:56 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:11:02 INFO: [Master] [Local integrity] Finished in 66.384s. Calculated metadata of 2248 files.
2022/08/26 11:11:10 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:11:10 INFO: [Master] [Local integrity] Finished in 0.162s. Calculated metadata of 2248 files.
2022/08/26 11:11:18 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:11:19 INFO: [Master] [Local integrity] Finished in 0.252s. Calculated metadata of 2248 files.
2022/08/26 11:11:25 INFO: [Worker worker1] [Integrity sync] Finished in 91.955s.
2022/08/26 11:11:26 INFO: [Worker worker2] [Integrity sync] Finished in 92.203s.
2022/08/26 11:11:27 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:11:27 INFO: [Master] [Local integrity] Finished in 0.270s. Calculated metadata of 2248 files.
2022/08/26 11:11:35 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:11:35 INFO: [Master] [Local integrity] Finished in 0.159s. Calculated metadata of 2248 files.
2022/08/26 11:11:43 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:11:43 INFO: [Master] [Local integrity] Finished in 0.281s. Calculated metadata of 2248 files.
2022/08/26 11:11:51 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:11:52 INFO: [Master] [Local integrity] Finished in 0.128s. Calculated metadata of 2248 files.
2022/08/26 11:12:00 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:12:00 INFO: [Master] [Local integrity] Finished in 0.139s. Calculated metadata of 2248 files.
2022/08/26 11:12:05 INFO: [Worker worker1] [Integrity check] Starting.
2022/08/26 11:12:05 INFO: [Worker worker1] [Integrity check] Finished in 0.021s. Received metadata of 2248 files. Sync not required.
2022/08/26 11:12:05 INFO: [Worker worker2] [Integrity check] Starting.
2022/08/26 11:12:05 INFO: [Worker worker2] [Integrity check] Finished in 0.013s. Received metadata of 2248 files. Sync not required.
```

The above is the result when using compression, which is enabled and recommend by default. However, even if compressed zip is too big or compression is disabled, it would still work (as shown in the use case below):
```
2022/08/26 11:04:40 INFO: [Worker] [Main] Connection from ('192.168.16.6', 53586)
2022/08/26 11:04:42 INFO: [Worker] [Main] Connection from ('192.168.16.5', 53488)
2022/08/26 11:04:48 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:04:48 INFO: [Master] [Local integrity] Finished in 0.162s. Calculated metadata of 2248 files.
2022/08/26 11:04:49 INFO: [Worker worker2] [Integrity check] Starting.
2022/08/26 11:04:49 INFO: [Worker worker2] [Integrity check] Finished in 0.004s. Received metadata of 34 files. Sync required.
2022/08/26 11:04:49 INFO: [Worker worker2] [Integrity sync] Starting.
2022/08/26 11:04:49 INFO: [Worker worker2] [Integrity sync] Files to create in worker: 2214 | Files to update in worker: 0 | Files to delete in worker: 0 | Files to receive: 0
2022/08/26 11:04:51 INFO: [Worker worker1] [Integrity check] Starting.
2022/08/26 11:04:51 INFO: [Worker worker1] [Integrity check] Finished in 0.004s. Received metadata of 34 files. Sync required.
2022/08/26 11:04:51 INFO: [Worker worker1] [Integrity sync] Starting.
2022/08/26 11:04:51 INFO: [Worker worker1] [Integrity sync] Files to create in worker: 2214 | Files to update in worker: 0 | Files to delete in worker: 0 | Files to receive: 0
2022/08/26 11:04:53 WARNING: [Local Server] [Main] Maximum zip size exceeded. Not all files will be compressed during this sync.
2022/08/26 11:04:55 WARNING: [Local Server] [Main] Maximum zip size exceeded. Not all files will be compressed during this sync.
2022/08/26 11:04:56 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:04:56 INFO: [Master] [Local integrity] Finished in 0.207s. Calculated metadata of 2248 files.
2022/08/26 11:05:04 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:05:04 INFO: [Master] [Local integrity] Finished in 0.172s. Calculated metadata of 2248 files.
2022/08/26 11:05:12 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:05:12 INFO: [Master] [Local integrity] Finished in 0.171s. Calculated metadata of 2248 files.
2022/08/26 11:05:20 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:05:21 INFO: [Master] [Local integrity] Finished in 0.205s. Calculated metadata of 2248 files.
2022/08/26 11:05:29 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:05:29 INFO: [Master] [Local integrity] Finished in 0.172s. Calculated metadata of 2248 files.
2022/08/26 11:05:37 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:05:37 INFO: [Master] [Local integrity] Finished in 0.161s. Calculated metadata of 2248 files.
2022/08/26 11:05:41 INFO: [Worker worker2] [Integrity sync] Finished in 51.869s.
2022/08/26 11:05:44 INFO: [Worker worker1] [Integrity sync] Finished in 52.868s.
2022/08/26 11:05:45 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:05:45 INFO: [Master] [Local integrity] Finished in 0.149s. Calculated metadata of 2248 files.
2022/08/26 11:05:53 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:05:53 INFO: [Master] [Local integrity] Finished in 0.136s. Calculated metadata of 2248 files.
2022/08/26 11:05:55 INFO: [Worker worker2] [Integrity check] Starting.
2022/08/26 11:05:55 INFO: [Worker worker2] [Integrity check] Finished in 0.016s. Received metadata of 1093 files. Sync required.
2022/08/26 11:05:55 INFO: [Worker worker2] [Integrity sync] Starting.
2022/08/26 11:05:55 INFO: [Worker worker2] [Integrity sync] Files to create in worker: 1155 | Files to update in worker: 0 | Files to delete in worker: 0 | Files to receive: 0
2022/08/26 11:05:56 INFO: [Worker worker1] [Integrity check] Starting.
2022/08/26 11:05:56 INFO: [Worker worker1] [Integrity check] Finished in 0.013s. Received metadata of 1093 files. Sync required.
2022/08/26 11:05:56 INFO: [Worker worker1] [Integrity sync] Starting.
2022/08/26 11:05:56 INFO: [Worker worker1] [Integrity sync] Files to create in worker: 1155 | Files to update in worker: 0 | Files to delete in worker: 0 | Files to receive: 0
2022/08/26 11:05:59 WARNING: [Local Server] [Main] Maximum zip size exceeded. Not all files will be compressed during this sync.
2022/08/26 11:06:00 WARNING: [Local Server] [Main] Maximum zip size exceeded. Not all files will be compressed during this sync.
2022/08/26 11:06:01 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:06:02 INFO: [Master] [Local integrity] Finished in 0.171s. Calculated metadata of 2248 files.
2022/08/26 11:06:10 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:06:10 INFO: [Master] [Local integrity] Finished in 0.174s. Calculated metadata of 2248 files.
2022/08/26 11:06:18 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:06:18 INFO: [Master] [Local integrity] Finished in 0.185s. Calculated metadata of 2248 files.
2022/08/26 11:06:26 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:06:26 INFO: [Master] [Local integrity] Finished in 0.262s. Calculated metadata of 2248 files.
2022/08/26 11:06:34 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:06:34 INFO: [Master] [Local integrity] Finished in 0.201s. Calculated metadata of 2248 files.
2022/08/26 11:06:41 INFO: [Worker worker1] [Integrity sync] Finished in 44.296s.
2022/08/26 11:06:42 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:06:43 INFO: [Master] [Local integrity] Finished in 0.179s. Calculated metadata of 2248 files.
2022/08/26 11:06:46 INFO: [Worker worker2] [Integrity sync] Finished in 50.724s.
2022/08/26 11:06:51 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:06:51 INFO: [Master] [Local integrity] Finished in 0.134s. Calculated metadata of 2248 files.
2022/08/26 11:06:53 INFO: [Worker worker1] [Integrity check] Starting.
2022/08/26 11:06:53 INFO: [Worker worker1] [Integrity check] Finished in 0.020s. Received metadata of 2137 files. Sync required.
2022/08/26 11:06:53 INFO: [Worker worker1] [Integrity sync] Starting.
2022/08/26 11:06:53 INFO: [Worker worker1] [Integrity sync] Files to create in worker: 111 | Files to update in worker: 0 | Files to delete in worker: 0 | Files to receive: 0
2022/08/26 11:06:57 INFO: [Worker worker1] [Integrity sync] Finished in 4.062s.
2022/08/26 11:06:59 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:06:59 INFO: [Master] [Local integrity] Finished in 0.130s. Calculated metadata of 2248 files.
2022/08/26 11:07:01 INFO: [Worker worker2] [Integrity check] Starting.
2022/08/26 11:07:01 INFO: [Worker worker2] [Integrity check] Finished in 0.024s. Received metadata of 2137 files. Sync required.
2022/08/26 11:07:01 INFO: [Worker worker2] [Integrity sync] Starting.
2022/08/26 11:07:01 INFO: [Worker worker2] [Integrity sync] Files to create in worker: 111 | Files to update in worker: 0 | Files to delete in worker: 0 | Files to receive: 0
2022/08/26 11:07:02 INFO: [Worker worker1] [Integrity check] Starting.
2022/08/26 11:07:03 INFO: [Worker worker1] [Integrity check] Finished in 0.108s. Received metadata of 2248 files. Sync not required.
2022/08/26 11:07:05 INFO: [Worker worker2] [Integrity sync] Finished in 4.164s.
2022/08/26 11:07:07 INFO: [Master] [Local integrity] Starting.
2022/08/26 11:07:07 INFO: [Master] [Local integrity] Finished in 0.137s. Calculated metadata of 2248 files.
2022/08/26 11:07:10 INFO: [Worker worker2] [Integrity check] Starting.
2022/08/26 11:07:10 INFO: [Worker worker2] [Integrity check] Finished in 0.021s. Received metadata of 2248 files. Sync not required.
```

## Tests

System tests (after being backported):
```
$ pytest -vv test_cluster/test_integrity_sync/
============================================================================================ test session starts ============================================================================================
platform linux -- Python 3.8.10, pytest-5.0.0, py-1.8.2, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.8.10', 'Platform': 'Linux-5.4.0-124-generic-x86_64-with-glibc2.29', 'Packages': {'pytest': '5.0.0', 'py': '1.8.2', 'pluggy': '0.13.1'}, 'Plugins': {'metadata': '1.10.0', 'html': '3.1.1', 'testinfra': '5.0.0', 'tavern': '1.2.2', 'pep8': '1.0.6', 'cov': '2.10.0', 'asyncio': '0.14.0'}}
rootdir: /home/selu/Git/wazuh-qa/tests/system
plugins: metadata-1.10.0, html-3.1.1, testinfra-5.0.0, tavern-1.2.2, pep8-1.0.6, cov-2.10.0, asyncio-0.14.0
collected 5 items                                                                                                                                                                                           

test_cluster/test_integrity_sync/test_integrity_sync.py::test_missing_file PASSED                                                                                                                     [ 20%]
test_cluster/test_integrity_sync/test_integrity_sync.py::test_shared_files PASSED                                                                                                                     [ 40%]
test_cluster/test_integrity_sync/test_integrity_sync.py::test_extra_files PASSED                                                                                                                      [ 60%]
test_cluster/test_integrity_sync/test_integrity_sync.py::test_extra_valid_files PASSED                                                                                                                [ 80%]
test_cluster/test_integrity_sync/test_integrity_sync.py::test_zip_size_limit PASSED                                                                                                                   [100%]

======================================================================================== 5 passed in 841.34 seconds =========================================================================================
```

### CODE REVIEW

File: framework/wazuh/core/cluster/cluster.py [hunk 1]
```diff
@@ -73,59 +64,30 @@ def check_cluster_config(config):
...
+        raise WazuhError(3004, "Port has to be an integer.")
+
     elif not 1024 < config['port'] < 65535:
         raise WazuhError(3004, "Port must be higher than 1024 and lower than 65535.")
 
     if len(config['nodes']) > 1:
         logger.warning(
             "Found more than one node in configuration. Only master node should be specified. Using {} as master.".
-            format(config['nodes'][0]))
+                format(config['nodes'][0]))
```
Comment (AdriiiPRodri) [inline framework/wazuh/core/cluster/cluster.py:83]:
```
f"Found more than one node in configuration. "
f"Only master node should be specified. Using {config['nodes'][0]} as master.")
```

File: framework/wazuh/core/cluster/cluster.py [hunk 2]
```diff
@@ -304,67 +266,107 @@ def get_ruleset_status(previous_status):
...
-            ko_files['shared'].pop(f, None)
+    try:
+        if failed_file in ko_files['missing']:
+            ko_files['missing'].pop(failed_file, None)
+        elif failed_file in ko_files['shared']:
+            if not exists:
+                ko_files['extra'][failed_file] = ko_files['shared'][failed_file]
+            ko_files['shared'].pop(failed_file, None)
+    except (KeyError, AttributeError, TypeError):
+        pass
```
Comment (AdriiiPRodri) [inline framework/wazuh/core/cluster/cluster.py:298]:
```suggestion
    with contextlib.suppress(KeyError, AttributeError, TypeError):
        if failed_file in ko_files['missing']:
            ko_files['missing'].pop(failed_file, None)
        elif failed_file in ko_files['shared']:
            if not exists:
                ko_files['extra'][failed_file] = ko_files['shared'][failed_file]
            ko_files['shared'].pop(failed_file, None)
```

File: framework/wazuh/core/cluster/cluster.py [hunk 3]
```diff
@@ -391,42 +393,71 @@ async def async_decompress_files(zip_path, ko_files_name="files_metadata.json"):
...
     ko_files : dict
         Paths (keys) and metadata (values) of the files listed in cluster.json.
     zip_dir : str
-        Full path to unzipped directory.
+        Full path to decompressed directory.
     """
+    ko_files = ''
+    compressed_data = b''
+    window_size = 1024*1024*10   # 10 MiB
+    decompress_dir = compress_path + 'dir'
```
Comment (AdriiiPRodri) [inline framework/wazuh/core/cluster/cluster.py:419]:
```suggestion
    window_size = 1024 * 1024 * 10
    decompress_dir = f'{compress_path}dir'
```

**Review by AdriiiPRodri — CHANGES_REQUESTED** [review]

**Review by AdriiiPRodri — APPROVED** [review]

**Comment by Selutario** [comment]
# Testing in a large environment
## Environment
- 1 Master node.
- 10 Worker nodes.
- 1000 agents.

## A. Compression level: 1

The level 1 is the default compression in the cluster, so these logs show the most common use case. 

We created 1070 groups and 2200 multigroups in the master node before starting `wazuh-clusterd`. Once started, the cheksum of all the files is calculated:
```
2022/09/06 11:45:15 INFO: [Master] [Local integrity] Finished in 4.740s. Calculated metadata of 4905 files.
```

Workers connect more or less at the same time and start the integrity task. We will follow the worker6 trace on master's logs to find out how it behaves:
```
2022/09/06 11:45:20 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_6] [Integrity check] Starting.
2022/09/06 11:45:20 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_6] [Integrity check] Finished in 0.009s. Received metadata of 534 files. Sync required.
2022/09/06 11:45:20 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_6] [Integrity sync] Starting.
2022/09/06 11:45:20 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_6] [Integrity sync] Files to create in worker: 4371 | Files to update in worker: 1 | Files to delete in worker: 0 | Files to receive: 0
2022/09/06 11:45:20 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_6] [Integrity sync] Compressing files to be synced in worker.
...
2022/09/06 11:46:34 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_6] [Integrity sync] Zip with files to be synced sent to worker.
2022/09/06 11:46:34 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_6] [Integrity sync] Finished sending files to worker.
2022/09/06 11:46:34 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_6] [Integrity sync] Finished in 73.667s.
```

The master syncs 4371 files with worker6 in 73.667 seconds. The zip of all those compressed files ends up with a size of 463 MB. The original size without compression would be close to 2100 MiB:
```
[root@ip-172-31-87-214 cluster]# ls -lh CLUSTER-Workload_benchmarks_metrics_B182_manager_9
total 463M
-rw-r--r-- 1 wazuh wazuh 463M sep  6 11:46 CLUSTER-Workload_benchmarks_metrics_B182_manager_9-1662464767.347322-bde508b14fab4ca9ab00823494d48ef4.zip
```

Please note that compression is now the most time-consuming part of the synchronization process, but it is executed in a child process of the cluster so that the parent can do other things in the meantime. By default, clusterd has only 2 child processes:
https://github.com/wazuh/wazuh/blob/3828669706ac1e13a6b824feb3635ff876c620f0/framework/wazuh/core/cluster/cluster.json#L103-L104
This means that the first two workers that start the Integrity sync task (worker6 and worker10 in these logs) won't have to wait for the compression to start:
```
2022/09/06 11:46:34 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_6] [Integrity sync] Finished in 73.667s.
2022/09/06 11:46:34 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_10] [Integrity sync] Zip with files to be synced sent to worker.
2022/09/06 11:46:34 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_10] [Integrity sync] Finished sending files to worker.
2022/09/06 11:46:34 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_10] [Integrity sync] Finished in 73.744s.
```

However, the other workers must wait for the child processes to be free, forming a queue in which the latter will wait for their own execution and for the execution of all the elements that have occurred before them:
```
2022/09/06 11:47:20 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_9] [Integrity sync] Finished in 119.816s.
2022/09/06 11:47:21 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_8] [Integrity sync] Finished in 120.437s.
...
2022/09/06 11:48:09 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_1] [Integrity sync] Finished in 168.216s.
2022/09/06 11:48:10 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_7] [Integrity sync] Finished in 169.027s.
...
2022/09/06 11:48:51 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_2] [Integrity sync] Finished in 210.259s.
2022/09/06 11:48:56 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Finished in 215.860s.
...
2022/09/06 11:49:28 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_3] [Integrity sync] Finished in 247.335s.
2022/09/06 11:49:37 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_4] [Integrity sync] Finished in 256.865s.
```

This time can be reduced by increasing the number of child processes available in the master's `wazuh-clusterd`. In any case, regardless of how long the synchronization takes (4.5 minutes in this worst case), it ends up without generating any errors.

The full log file can be downloaded [here](https://github.com/wazuh/wazuh/files/9498110/master1_cluster.log).

**Comment by Selutario** [comment]
# Testing in a large environment
## Environment
- 1 Master node.
- 10 Worker nodes.
- 1000 agents.

## B. Compression level: 0

I set the compression level to 0 and restarted all nodes in the cluster, taking advantage of [this bug in 4.3.7](https://github.com/wazuh/wazuh/issues/14814) that causes all multigroups to be removed from workers when stopping the service.

This forces the master to send the 2200 multigroups, whose collective size is still very close to 2100 MiB, back to all the workers. Also, since the compression level is 0 (off), the parent process of `wazuh-clusterd` on the master will send much larger files than in the [previous test](https://github.com/wazuh/wazuh/pull/14746#issuecomment-1238261846).

Let's follow the worker5 trace, for example:
```
2022/09/06 11:52:19 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity check] Starting.
2022/09/06 11:52:19 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity check] Finished in 0.022s. Received metadata of 2691 files. Sync required.
2022/09/06 11:52:19 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Starting.
2022/09/06 11:52:19 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Files to create in worker: 2214 | Files to update in worker: 0 | Files to delete in worker: 0 | Files to receive: 0
2022/09/06 11:52:19 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Compressing files to be synced in worker.
2022/09/06 11:54:44 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Main] Command received: b'cancel_task'
2022/09/06 11:54:44 ERROR: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Main] The task was canceled due to the following error on the remote node: Error 3039 - Timeout while waiting to receive a file
2022/09/06 11:54:45 ERROR: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Main] Error sending files information: Error 3027 - Unknown received task name: 373fa4b8-b491-4642-96ab-498770a24528
2022/09/06 11:54:45 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Main] Decreasing sync size limit to 611.98 MB.
2022/09/06 11:54:46 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Finished sending files to worker.
2022/09/06 11:54:46 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Finished in 146.479s.
...
2022/09/06 11:54:53 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity check] Starting.
2022/09/06 11:54:53 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity check] Finished in 0.612s. Received metadata of 2691 files. Sync required.
2022/09/06 11:54:53 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Starting.
2022/09/06 11:54:53 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Files to create in worker: 2214 | Files to update in worker: 0 | Files to delete in worker: 0 | Files to receive: 0
2022/09/06 11:54:53 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Compressing files to be synced in worker.
2022/09/06 11:54:59 WARNING: [Local Server] [Main] Maximum zip size exceeded. Not all files will be compressed during this sync.
2022/09/06 11:56:33 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Zip with files to be synced sent to worker.
2022/09/06 11:56:33 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Main] Increasing sync size limit to 764.97 MB.
2022/09/06 11:56:33 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Finished sending files to worker.
2022/09/06 11:56:33 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Finished in 99.861s.
...
2022/09/06 11:56:43 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity check] Starting.
2022/09/06 11:56:43 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity check] Finished in 0.702s. Received metadata of 3316 files. Sync required.
2022/09/06 11:56:43 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Starting.
2022/09/06 11:56:43 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Files to create in worker: 1589 | Files to update in worker: 0 | Files to delete in worker: 0 | Files to receive: 0
2022/09/06 11:56:43 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Compressing files to be synced in worker.
2022/09/06 11:56:59 WARNING: [Local Server] [Main] Maximum zip size exceeded. Not all files will be compressed during this sync.
2022/09/06 11:58:13 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Zip with files to be synced sent to worker.
2022/09/06 11:58:13 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Main] Increasing sync size limit to 956.22 MB.
2022/09/06 11:58:13 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Finished sending files to worker.
2022/09/06 11:58:13 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Finished in 89.962s.
...
2022/09/06 11:58:24 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity check] Starting.
2022/09/06 11:58:25 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity check] Finished in 0.673s. Received metadata of 4105 files. Sync required.
2022/09/06 11:58:25 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Starting.
2022/09/06 11:58:25 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Files to create in worker: 800 | Files to update in worker: 0 | Files to delete in worker: 0 | Files to receive: 0
2022/09/06 11:58:25 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Compressing files to be synced in worker.
2022/09/06 12:00:26 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Zip with files to be synced sent to worker.
2022/09/06 12:00:26 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Finished sending files to worker.
2022/09/06 12:00:26 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity sync] Finished in 120.952s.
...
2022/09/06 12:00:33 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity check] Starting.
2022/09/06 12:00:33 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B182_manager_5] [Integrity check] Finished in 0.086s. Received metadata of 4905 files. Sync not required.
```

When the zip to be sent is too large to be synced correctly (`timeoutError`), the maximum zip size is dynamically adapted, eventually allowing the Integrity sync task to succeed as shown in the logs above. 

This can be useful especially in those environments where, despite compressing files, the master is still not able to send it to the worker on time.

The full log file can be downloaded [here](https://github.com/wazuh/wazuh/files/9498400/master2_cluster.log).

**Pull request closed without merging**